src/sentry/static/sentry/app/views/organizationIntegrations/integrationListDirectory.tsx

```diff
@@ -289,6 +316,26 @@ export class OrganizationIntegrations extends AsyncComponent<
     });
   };
 
+  onCategorySelect = async ({value: category}: {value: string}) => {
```

Comment:
@NisanthanNanthakumar  I am confused why this is async and why you are using setState with a callback function

Reply:
@scefali haha, it does NOT need `async`. 

I'm using the callback function because I want the category selector to update with the new value and not hang while the new list of filtered integrations is being calculated.